### DISCUSSION TITLE
Results from linkedin auth are null besides name

### DISCUSSION THREAD

**Discussion by ludersGabriel** [nextauthjs/next-auth#1541] [Help] [answered]
**Your question**

Is linkedin auth bugged for email?
**What are you trying to do**

I'm trying to recover the email from linkedin with the scope 'r_emailaddress r_liteprofile', but the results are null. Linkedin's login confirmation page shows that the app is requesting email and I have another website that uses the same linkedin app that works, so I'm starting to believe that this might be a bug.
**Reproduction**

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment]
It's probably because the built-in LinkedIn provider does not set the email by default:
https://github.com/nextauthjs/next-auth/blob/776b9480da7d0b51bf3b1358be61fa0490126011/src/providers/linkedin.js#L20

try overriding the profile callback and return e-mail for your provider. If it is something that can be fetched in one pass (meaning that adding a scope), I think the default LinkedIn provider behavior could also be tweaked to get the e-mail by default.

**Comment by mattfranciswork0** [comment]
Hey @ludersGabriel , were you able to solve this?

**Reply by ludersGabriel** [reply]
yeah! I did what @balazsorban44 said. The code looks like the following inside my [nextauth...]'s providers

![image](https://user-images.githubusercontent.com/57847215/123629071-35ceda80-d7ea-11eb-8f86-e676504b911b.png)

**Reply by mattfranciswork0** [reply]
Thank you so much for your response! Appericiate this, must've been challenging to crawl through their docs!

If you mind sharing, could you send the code snippet too (rather than a screenshot!) ? Just to make it easier to implement :)!

**Reply by ludersGabriel** [reply]
Sure

```
{
    id: "linkedin",
    name: "LinkedIn",
    type: "oauth",
    version: "2.0",
    scope: "r_emailaddress r_liteprofile",
    params: {
      grant_type: "authorization_code",
      client_id: process.env.LINKEDIN_ID,
      client_secret: process.env.LINKEDIN_SECRET,
    },
    accessTokenUrl: "https://www.linkedin.com/oauth/v2/accessToken",
    authorizationUrl:
      "https://www.linkedin.com/oauth/v2/authorization?response_type=code",
    profileUrl:
      "https://api.linkedin.com/v2/me?projection=(id,localizedFirstName,localizedLastName)",
    profile: async (profile, tokens) => {
      const res = await fetch(
        `https://api.linkedin.com/v2/emailAddress?q=members&projection=(elements*(handle~))&oauth2_access_token=${tokens.access_token}`
      )
      const data = await res.json()
      const email = data.elements[0]["handle~"].emailAddress
      return {
        id: profile.id,
        name: profile.localizedFirstName + " " + profile.localizedLastName,
        email,
        image: null,
      }
    },
    clientId: process.env.LINKEDIN_ID,
    clientSecret: process.env.LINKEDIN_SECRET,
  },

```

**Reply by mattfranciswork0** [reply]
You are amazing! Thank you so much, means a lot!

**Reply by balazsorban44** [reply]
We take PRs with documentation improvement suggestions. 😉

**Reply by ludersGabriel** [reply]
I believe he meant linkedin's documentation which is a nightmare haha

**Reply by balazsorban44** [reply]
Ah, sorry. Hopefully haven't had the chance with that yet. 😅

**Reply by mattfranciswork0** [reply]
Exactly what @ludersGabriel  said haha. The next-auth documentation is great as it is :).

**Comment by balazsorban44** [comment, marked as answer]
see https://github.com/nextauthjs/next-auth/pull/1329

https://next-auth.js.org/configuration/providers#using-a-custom-provider

**Reply by mattfranciswork0** [reply]
You're a life saver, thank you!!!
